Hash/Hash.cs

```diff
@@ -0,0 +1,158 @@
+        }
+
+        public static string SHA512Hash(string text)
+        {
+            
+            SHA512 shaM = new SHA512Managed();
+            shaM.ComputeHash(ASCIIEncoding.ASCII.GetBytes(text));
+            byte[] result = shaM.Hash;
+
+            return Hash.StrAppend(result);
+        }
+
+        public static string MD5Hash(string text)
+        {
+            MD5 md5 = new MD5CryptoServiceProvider();
+            md5.ComputeHash(ASCIIEncoding.ASCII.GetBytes(text));
+            byte[] result = md5.Hash;
+
+            return Hash.StrAppend(result);
+        }
+
+        private static async Task<string> ReadRequestBodyAsync(HttpRequest request)
+        {
+            string bodyAsText = string.Empty;
+            if (request.Body.CanSeek)
+            {
+                request.Body.Seek(0, SeekOrigin.Begin);
+                using var reader = new StreamReader(request.Body);
+                bodyAsText = await reader.ReadToEndAsync();
+                request.Body.Seek(0, SeekOrigin.Begin);
+            }
+            return bodyAsText;
+        }
+
+        [FunctionName("Hash")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)] HttpRequest req,
+            ILogger log)
+        {
+            log.LogInformation($"C# HTTP trigger function processed a request. RequestQuery={req.Body}");
+            var identity = req.HttpContext?.User?.Identity;
+            log.LogInformation("IsAuthenticated: {isAuthenticated}", identity?.IsAuthenticated);
+            log.LogInformation("Identity name: {name}", identity?.Name);
+            log.LogInformation("AuthenticationType: {authenticationType}",
+                identity?.AuthenticationType);
+
+            var success = true;
+            string json = string.Empty;
+            string message = req.Query["message"];
+            string hashtype = req.Query["hashtype"];
+            string error = "Please pass a hashtype (md5/sha256/sha512/rsa) and the message which you want to encode in the request body";
+            string responseMessage = string.Empty;
+            var requestBodyContent = await Hash.ReadRequestBodyAsync(req);
+            message = requestBodyContent != string.Empty ? requestBodyContent : message;
+
+            try
+            {
+                switch (hashtype)
+                {
+                    case "sha256":
+                        responseMessage = Hash.SHA256Hash(message);
+                        break;
+                    case "sha512":
+                        responseMessage = Hash.SHA512Hash(message);
+                        break;
+                    case "rsa":
+                        responseMessage = Hash.RSA(message);
+                        break;
+                    default:
+                        responseMessage = Hash.MD5Hash(message);
+                        break;
+                }
+
+                // if you wanna json return

```

Comment:
commented code

Reply:
A cunning design. Hypothetically, it may be needed by those who will convert their answer to json)